eng/common/scripts/Helpers/PSModule-Helpers.ps1

```diff
@@ -0,0 +1,97 @@
+$DefaultPSRepositoryUrl = "https://www.powershellgallery.com/api/v2"
+$global:CurrentUserModulePath = ""
+
+function Update-PSModulePath()
+{
+  # Information on PSModulePath taken from docs
+  # https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_psmodulepath
+
+  # Information on Az custom module paths on hosted agents taken from
+  # https://github.com/microsoft/azure-pipelines-tasks/blob/c9771bc064cd60f47587c68e5c871b7cd13f0f28/Tasks/AzurePowerShellV5/Utility.ps1
+
+  if ($IsWindows) {
+    $hostedAgentModulePath = $env:SystemDrive + "\Modules"
+    $moduleSeperator = ";"
+  } else {
+    $hostedAgentModulePath = "/usr/share"
+    $moduleSeperator = ":"
+  }
+  $modulePaths = $env:PSModulePath -split $moduleSeperator
+
+  # Remove any hosted agent paths (needed to remove old default azure/azurerm paths which cause conflicts)
+  $modulePaths = $modulePaths.Where({ !$_.StartsWith($hostedAgentModulePath) })
+
+  # Add any "az_" paths from the agent which is the lastest set of azure modules
+  $AzModuleCachPath = (Get-ChildItem "$hostedAgentModulePath/az_*" -Attributes Directory) -join $moduleSeperator
+  if ($AzModuleCachPath -and $env.PSModulePath -notcontains $AzModuleCachPath) {
+    $modulePaths += $AzModuleCachPath
+  }
+
+  $env:PSModulePath = $modulePaths -join $moduleSeperator
+
+  # Find the path that is under user home directory
+  $homeDirectories = $modulePaths.Where({ $_.StartsWith($home) })
+  if ($homeDirectories.Count -gt 0) {
+    $global:CurrentUserModulePath = $homeDirectories[0]
+    if ($homeDirectories.Count -gt 1) {
+      Write-Verbose "Found more then one module path starting with $home so selecting the first one $global:CurrentUserModulePath"
```

Comment:
Here and below are other places I'd use the curly braces to be on the safe side.